Assets/Scripts/Background/BeyondBorderResize.cs

```diff
@@ -0,0 +1,118 @@
+using UnityEngine;
+
+public class BeyondBorderResize : MonoBehaviour
+{
+    private float cameraHeight;
+    private float cameraWidth;
+    private float spriteWidth;
+    private float spriteHeight;
+    private float sizeFactorWidth;
+    private float sizeFactorHeight;
+    private float borderWidth;
+    private float borderEdge;
+    private float cameraEdge;
+    private float localXScale;
+    private Vector2 currentSize;
+    private SpriteRenderer fillerSprite;
+
+    public bool right;  // true if filling the space for the right side of the screen, false if otherwise
+    public SpriteRenderer border;  // variable to store border sprite
+    
+    
+    void Awake()
+    {
+        cameraHeight = Camera.main.orthographicSize * 2;
+        cameraWidth = cameraHeight * Camera.main.aspect;
+        currentSize = new Vector2(cameraWidth, cameraHeight);
+        localXScale = transform.localScale.x;
+        borderWidth = border.size.x * border.transform.localScale.x;
+
+        if (!right) {
+            borderEdge = border.transform.position.x - (borderWidth / 2);
+            cameraEdge = Camera.main.transform.position.x - (cameraWidth / 2);
+        } else {
+            borderEdge = border.transform.position.x + (borderWidth / 2);
+            cameraEdge = Camera.main.transform.position.x + (cameraWidth / 2);
+        }
+        
+        fillerSprite = GetComponent<SpriteRenderer>();
+
+        // If background + edges combined are already wider than the camera, no need to resize sprite.
+        if ((right && borderEdge < cameraEdge) || (!right && borderEdge > cameraEdge)) {
+            spriteWidth = fillerSprite.size.x;
+            spriteHeight = fillerSprite.size.y;
+
+            // The sizing factor is calculated, with the sprite renderer transformed to fill an empty side of the screen.
+            if (!right) {
+                sizeFactorWidth = Mathf.Abs(borderEdge - cameraEdge) / spriteWidth;
+            } else {
+                sizeFactorWidth = Mathf.Abs(cameraEdge - borderEdge) / spriteWidth;
+            }
+
+            sizeFactorHeight = cameraHeight / spriteHeight;
+            transform.position = new Vector2((borderEdge + (cameraEdge)) / 2, Camera.main.transform.position.y);
+            fillerSprite.size = new Vector2(spriteWidth * sizeFactorWidth / localXScale, spriteHeight * sizeFactorHeight / localXScale);
+        }
+    }
+
+
+    // ChangeSize
+    // This function updates the empty right/left space when the camera size has changed (just like the Awake function).
+    private void ChangeSize()
+    {
+        float newSizeFactorWidth;
+        float newSizeFactorHeight;
+
+        if (!right) {
+            cameraEdge = Camera.main.transform.position.x - (cameraWidth / 2);
+        } else {
+            cameraEdge = Camera.main.transform.position.x + (cameraWidth / 2);
+        }
+        
+        if ((right && borderEdge < cameraEdge) || (!right && borderEdge > cameraEdge)) { 

```

Comment:
It looks like this if statement duplicates what is in `Awake`. You should extract a method that contains this code to reduce code duplication.

Reply:
Got it; all duplicated code between Awake and "UpdateBackgroundFillSize" has been placed inside "UpdateBackgroundFillSize", with Awake calling said function.